### PR TITLE
Do not log SuppressedError to GitHub CI

### PR DESCRIPTION
**Pull request by Electroid**
### What does this PR do?

* Prevents `SuppressedError` from being annotated in GitHub CI

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

Automated tests.

### CODE REVIEW

File: src/bun.js/javascript.zig [hunk 1]
```diff
@@ -3078,6 +3084,7 @@ pub const VirtualMachine = struct {
                 defer source_url.deinit();
                 const file = bun.path.relative(dir, source_url.slice());
                 const func = frame.function_name.toUTF8(allocator);
+                defer func.deinit();
```
Comment (jdalton) [inline src/bun.js/javascript.zig:3083]: Nice catch!

**Review by jdalton — APPROVED** [review]

**Comment by paperclover** [comment]
Why? SuppressedError is still an Error. And an uncaught error is still an error.

**Comment by jdalton** [comment]
@Electroid  The intent is to allow devs to suppress errors from Github Actions. 
The use of [SuppressedError](https://github.com/tc39/proposal-explicit-resource-management?tab=readme-ov-file#the-suppressederror-error) is a neat way to do that though seems a little off-of-center from its intended use. Can we take inspiration from that and sniff the error `name` (for something similar) or maybe make it configurable of which error names are not to be annotated.

**Review by paperclover — CHANGES_REQUESTED** [review]
The intent of this PR is make it possible to reduce noise in CIs by making a suppressable error type. that is not what SuppressedError does.

Do not use `SuppressedError` for this purpose. It can be thrown by built-in syntax from the using spec as linked by John-David. **These errors should always be logged in CI and the terminal**.

In my opinion we shouldnt have any error for this, because the fact the error is logged actually a different bug in bun, and this just hides it.

**Pull request closed without merging**